### PR TITLE
Validation of Type Definitions

### LINKED ISSUE #213: Narrowing interface fields in an implementation fails
Consider the following example:

```ts
import { Neo4jGraphQL } from '@neo4j/graphql';

new Neo4jGraphQL({
  typeDefs: /* GraphQL */ `
    interface Vehicle {
      id: ID!
      color: String # NOTE: 'color' is optional on the interface
    }

    type Car implements Vehicle {
      id: ID!
      color: String! # NOTE: 'color' is mandatory on the type, which should be okay
    }

    type Query {
      cars: [Vehicle!]!
    }
  `,
});
```
Here, `color` is defined as an optional string in the Vehicle interface. In the Car implementation, the `color` field is mandatory. In GraphQL an optional type in an interface may be implemented as mandatory. Trying to do this with Neo4jGraphQL fails with:

```
Error: type Car does not implement interface Vehicle correctly
```

**System:**

"@neo4j/graphql": "1.0.1",

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

A second crack at validation of type definitions. Seems to work for all cases, but some users may have to disable the functionality, for instance, if they are using one of our generated types in their type definitions, as this will fail.

Additionally, removed the `ID` scalar and switches it to the same functionality that `Int` and `Float` now use.

# Issue

Fixes #213, as well as minimises the amount of checking we do ourselves, and prevents users from passing in their own `Point`, `PointInput`, `DateTime`, etc.

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Review by danstarns — CHANGES_REQUESTED** [review]

File: packages/graphql/src/schema/validation/validate-document.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                        ...f,
+                        arguments: f.arguments?.filter((argument) => {
+                            const getArgumentType = (type) => {
+                                if (["NonNullType", "ListType"].includes(type.kind)) {
+                                    return getArgumentType(type.type);
+                                }
+                                return type.name.value;
+                            };
+                            const type = getArgumentType(argument.type);
+                            const match = /(?<nodeName>.+)(?:CreateInput|UpdateInput|Where)/gm.exec(type);
```
[the remark this answers is not in the document]
Comment (darrellwarde) [reply packages/graphql/src/schema/validation/validate-document.ts]: Honestly it was just a selection of types which might be common use. I've added the sort type, but we can tweak this as we go along!

File: packages/graphql/src/schema/make-augmented-schema.ts [hunk 2]
```diff
@@ -1275,9 +1291,30 @@ function makeAugmentedSchema(
...
+            seen[n] = n;
+
+            return true;
+        }),
+    };
+
     const schema = makeExecutableSchema({
         ...schemaDefinition,
-        typeDefs: generatedTypeDefs,
+        typeDefs: print(parsedDoc),
```
[the remark this answers is not in the document]
Comment (darrellwarde) [reply packages/graphql/src/schema/make-augmented-schema.ts]: I believe this was from your branch which I branched from... 😉  But sure, removed that now.

[the remark this answers is not in the document]
Comment (danstarns) [reply packages/graphql/src/schema/make-augmented-schema.ts]: ahahaha oops 🤐

**Review by danstarns — APPROVED** [review]
Lets get it in :)

**Pull request merged**